app/controllers/project/Projects.scala

```diff
@@ -194,9 +197,11 @@ class Projects @Inject()(stats: StatTracker,
   def show(author: String, slug: String) = ProjectAction(author, slug) async { request =>
     val project = request.project
     implicit val r = request.request
-    val rootPages = null // TODO root pages
-    val pageCount: Int = 0 // TODO page count
-    this.stats.projectViewed(request => Ok(views.pages.view(project, rootPages, project.p.homePage, pageCount)))(request)
+
+    projects.queryProjectPages(project.p) flatMap { pages =>
+      val pageCount = pages.size + pages.values.map(_.size).sum
+      this.stats.projectViewed(request => Ok(views.pages.view(project, pages, project.p.homePage, None, pageCount)))(request)
```

Comment:
Any reason request is no longer marked as implicit, but instead passed in explicitly here?